relate/src/main/scala/com/lucidchart/relate/ColReader.scala

```diff
@@ -59,7 +59,8 @@ object ColReader {
   implicit val boolReader: ColReader[Boolean] = optReader((col, rs) => rs.getBoolean(col))
   implicit val byteArrayReader: ColReader[Array[Byte]] = optReader((col, rs) => rs.getBytes(col))
   implicit val byteReader: ColReader[Byte] = optReader((col, rs) => rs.getByte(col))
-  implicit val dateReader: ColReader[Date] = optReader((col, rs) => rs.getDate(col))
+  implicit val dateReader: ColReader[Date] = optReader((col, rs) => rs.getTimestamp(col)).map(timestamp => new Date(timestamp.getTime))
```

Comment:
While we are here, we might as well add a `ColReader` for java.sql.Timestamp.

Or maybe do that instead of adding a `ColReader` for `java.util.Date`? I think either path would be fine.